aten/src/ATen/native/cpu/BinaryOpsKernel.cpp

```diff
@@ -523,6 +523,19 @@ void mse_kernel(TensorIterator& iter) {
   });
 }
 
+bool equal_kernel(TensorIterator& iter) {
+  std::atomic<bool> equal{true};
+  AT_DISPATCH_ALL_TYPES_AND_COMPLEX_AND(kBool, iter.dtype(), "equal_cpu", [&]() {
+    cpu_kernel(iter,
+      [&](scalar_t x, scalar_t y) -> scalar_t {
```

Comment:
Shouldn't specify a return type here (It's causing build errors). Capture can be more specific `[&equal]`

Reply:
It's weird that the unit test will fail when not specify a return type here. But test will be passed with a return type.